### DISCUSSION TITLE
Modifying Properties in a Local iModel

### DISCUSSION THREAD

**Discussion by Nhillus** [iTwin/itwinjs-core#6632] [Q&A] [answered]
Hi again! Thanks for your previous assistance. After applying your tips, I conducted some tests on the local iModel that I downloaded using the briefcase. However, I'm encountering issues with modifying properties following those changes sugested.

![image](https://github.com/iTwin/itwinjs-core/assets/46725871/f5dc52ef-be49-4070-98d3-0bc155036b2f)

To allow editing, I enabled isPropertyEditingEnabled to true in the viewer, which made the input box appear, but it doesn't actually modify the property when I press enter. I'm seeking guidance in this part.

From what I understand, I could possibly resolve this by using an RPC interface to communicate with the app's backend, perhaps using updateElement as suggested by another maintainer. Thank you again for posting those links!. In my main project, I used RpcInterface from @itwin/core-common to create a custom interface, and I think this might be the correct approach to locally modify properties. Would this method be appropriate? or maybe its a better aproach?

Additionally, I have a question regarding what exactly constitutes an "element" within this context. For my tests, I used a snapshot to access the iModel that i previously download and printed it, but I'm not seeing anything related to bis.elements or other tables that exist in the BIM file.

![elementsImodel](https://github.com/iTwin/itwinjs-core/assets/46725871/278b343e-a703-4c36-a0e6-dfbd743e1578)

It seems like I'm accessing iTwin information rather than the iModel specifics. Could you clarify this for me?

And again thanks for your help!

**Comment by grigasp** [comment, marked as answer]
> To allow editing, I enabled isPropertyEditingEnabled to true in the viewer, which made the input box appear, but it doesn't actually modify the property when I press enter. I'm seeking guidance in this part.

The property grid component doesn't handle updates - you'll have to do that yourself. Assuming you're using the widget from `@itwin/property-grid-react`, you can supply the [`onPropertyUpdated`](https://github.com/iTwin/viewer-components-react/blob/master/packages/itwin/property-grid/src/components/PropertyGridContent.tsx#L58) callback to be notified about the change in the UI, where you can initiate the actual change in the iModel.

> Additionally, I have a question regarding what exactly constitutes an "element" within this context. For my tests, I used a snapshot to access the iModel that i previously download and printed it, but I'm not seeing anything related to bis.elements or other tables that exist in the BIM file.
> It seems like I'm accessing iTwin information rather than the iModel specifics. Could you clarify this for me?

I'm sure what you expect to see there, but the `IModelDb` object doesn't store element information - it's queried from the db on demand when you request a specific element. In your screenshot there are members like `elements`, `models`, `views` and similar which provide you a way to access different kinds of elements.

**Reply by Nhillus** [reply]
Oh thanks i got it!! i archieve a edit on a element, but having some problems with editing a specific element i should get the element that i want by the source id rigth? a lot of my elements have the same federationGuid if anyone have Any way or tips in this part I would appreciate it Thanks again!

nvm it was a doc guid asociated ill mark the previous anwser as correct thanks again!
